app/javascript/helpers/Axios.jsx

```diff
@@ -1,5 +1,35 @@
 import axios from 'axios';
 
+// Merges custom params with search params.
+function mergeConfigParams(params, configParams = { }) {
+  Object.entries(configParams).forEach(
+    (param) => {
+      const key = param.at(0);
+      const val = param.at(1);
+      if (val) {
+        params.set(key, val);
+      }
+    },
+  );
+
+  return params;
+}
+
+// Filters out unwatned serach params.
+function filterParams(params) {
+  const allowedParams = ['search', 'sort[column]', 'sort[direction]'];
+  const isAllowed = (val) => allowedParams.includes(val);
+
+  const keys = Array.from(params.keys());
+  keys.forEach((key) => {
+    if (!isAllowed(key)) {
+      params.delete(key);
+    }
+  });
+
+  return params;
+}
```

Comment:
`filterParams(params: URLSearchParams): URLSearchParams`
This function will be responsible for:
1. Taking in a list of serach `params`.
2. Remove any param that is not allowed by iterating over all param names and check their legitimately with a predefined list of allowed param names `allowedParams`.
-> The end result will be search params that are allowed only to be passed in the requests.
This prevents registering and  sending random search params i.e. for `?foo=foo&bar=bar&search=GL` will be treated as `?search=GL` according to this list of allowed params.
This function then filter out serach params.